JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesSparql.java

```diff
@@ -480,4 +841,68 @@ private List<String> getInstances(SelectQuery instanceSelectQuery, String placeh
 
 		return instanceIRIs;
 	}
+
+	protected void initTS(String timeSeriesIRI, List<String> dataIRI, String dbURL, String timeUnit, Iri type, String durationIRI, Double duration, String unit){
```

Comment:
is this still needed? you are not matching the unit by string anymore?

Reply:
this is called from deleteTimeSeries in TimeSeriesClient

Reply:
Do you need to check the unit? aren't you just adding what was in the KG previously? if it's needed shouldn't it be matched with the full IRI?

Reply:
> Do you need to check the unit? aren't you just adding what was in the KG previously? if it's needed shouldn't it be matched with the full IRI?

at the moment its being called from deleteTimeSeries but the design should not be based on that. In future if someone wants to use it directly, removing the checks eliminate that usage. 
Okay, I'll match with the full IRI.

Reply:
resolved